### ISSUE TITLE
GetOperation should succeed on local function with block and expression bodies

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#24650]
In the test below, `GetOperation` should not return `null` since the semantic model allows querying that piece of syntax.

```C#
        [Fact]
        [WorkItem(24647, "https://github.com/dotnet/roslyn/issues/24647")]
        public void Repro24647()
        {
            var comp = CreateStandardCompilation(@"
class Program
{
    static void Main(string[] args)
    {
        void Func() { } => new object();
    }
}");
            var tree = comp.SyntaxTrees.Single();
            var model = comp.GetSemanticModel(tree, ignoreAccessibility: false);
            var creation = tree.GetRoot().DescendantNodes().OfType<ObjectCreationExpressionSyntax>().Single();

            var operation = model.GetOperation(creation);
            Assert.Null(operation); // we didn't bind the expression body, but should. See issue https://github.com/dotnet/roslyn/issues/24650

            var info = model.GetTypeInfo(creation);
            Assert.Equal("System.Object", info.Type.ToTestDisplayString());
            Assert.Equal("System.Object", info.ConvertedType.ToTestDisplayString());
        }
```

Follow-up to https://github.com/dotnet/roslyn/issues/24647

**Comment by 333fred** [comment]
So, the issue here is that because the binder does not bind the arrow expression, it's not part of the IOperation tree. `SemanticModel` will bind for error recovery, while it was a design decision that IOperation specifically does not look at the error recovery info to ensure there's only one tree.

@AlekseyTs, @jcouv mentioned that you thought the binder should actually bind this during initial bind, as opposed to falling back to error recovery. Could you elaborate?

**Comment by 333fred** [comment]
Fixed by #25782.
